src/FormBuilder/resources/js/form-builder/src/blocks/fields/amount/inspector/index.tsx

```diff
@@ -76,6 +77,30 @@ const Inspector = ({attributes, setAttributes}) => {
     const isRecurringSupported = enabledGateways.some((gateway) => gateway.supportsSubscriptions);
     const isRecurring = isRecurringSupported && recurringEnabled;
 
+    const [levelsOptions, setLevelsOptions] = useState<OptionProps[]>(
+        levels.map((level) => ({
+            label: formatCurrencyAmount(level),
+            value: level,
+            checked: defaultLevel === level,
+        }))
+    );
+
+    const checkedLevelOption = levelsOptions.filter((option) => option.checked);
+    if (!!checkedLevelOption && checkedLevelOption.length === 1) {
+        setAttributes({defaultLevel: checkedLevelOption[0].value});
+    } else if (levelsOptions.length > 0) {
+        levelsOptions[0].checked = true;
+    }
+
+    const setNewLevels = (levelsOptions: OptionProps[]) => {
+        const newLevels = levelsOptions.filter((option) => option.value).map((option) => option.value);
+        if (JSON.stringify(newLevels) !== JSON.stringify(levels)) {
+            setAttributes({levels: newLevels});
+        }
+    };
+
+    setNewLevels(levelsOptions); //Update levels when the reorder feature is used
+
```

Comment:
@glaubersilva I'm seeing a few issues here. It looks like there are side effects happening that will be triggered during every render/re-render.  This _could_ be prevented by putting the functions being fired inside a `useEffect` hook. 

Also, from first glance, i'm a little confused why we have both 
`setLevelsOptions` and `setNewLevels`.  If levelOptions is just converting the raw levels to an option structure, I would elevate that into a simple conversion function instead of trying to store it in state.  We should just be left with `setNewLevels` and `setDefaultLevel` callbacks.  The drag and drop should trigger a prop in the Options component like `onChange` and/or `setOptions` and therefore call `setNewLevels`.

Reply:
Now I'm using the `useEffect` to prevent re-render and I also changed the names to avoid confusion. The `useEffect` also updates the options on the screen when it gets recorded. About the `useState`, we need that because the Option component requires that to be able to update the options on the screen.